### DISCUSSION TITLE
anonymous function in javascript

### DISCUSSION THREAD

**Discussion by patrickwasp** [zauberzeug/nicegui#798] [Q&A] [answered]
### Question

I'd like to customize the legend label format in a Highcharts chart. 

In javascript this would be done using an anonymous function as the value for the `labelFormatter` key. How would I write this for nicegui?
```javascript
    legend: {
        labelFormatter: function () {
            return this.name + ' (click to hide)';
        }
    },
```

https://jsfiddle.net/gh/get/library/pure/highcharts/highcharts/tree/master/samples/highcharts/legend/labelformatter/

using triple quotes doesn't generate the correct result.
```python
legend = {
   "labelFormatter": """function() {return this.name + ' (click to hide)';}"""
}
```

`a.labelFormatter.call is not a function`

**Comment by falkoschindler** [comment, marked as answer]
Hi @patrickwasp!
Currently it is not possible to include callables in chart options. (I wish it was and I'm open for suggestions how to make it possible!)
As a workaround you can modify the chart later via `ui.run_javascript`, e.g. when the client connects:
```py
async def set_format():
    await ui.run_javascript('''
        getElement(%d).chart.update({
            legend: {
                labelFormatter: function() {
                    return this.name + ' (click to hide)';
                },
            },
        });
    ''' % chart.id, respond=False)
app.on_connect(set_format)
```

**Reply by EdwardPinus** [reply]
hello, i ask you a question: what happened run_javascript? thanks for the time

**Reply by falkoschindler** [reply]
@EdwardPinus Sorry, I don't understand your question. Can you elaborate?

**Reply by EdwardPinus** [reply]
Sorry, what do you do mean run_javascript?

**Reply by falkoschindler** [reply]
I'm referring to the function `ui.run_javascript`: https://nicegui.io/documentation/run_javascript
Not sure if this answers your question.

**Comment by jewalky** [comment]
In case anyone stumbles upon this in 2025, since Falko's answer there was discovered a way to make it possible and now more correct answer would be to use syntax like so:
```python
                "tooltip": {
                    ":formatter": "function() { return this.point.name + ': € ' + this.y.toFixed(2) }"
                }
```
(and in general by prepending `:` to options that need to be evaluated on JS side)

Exact reason this works: https://github.com/zauberzeug/nicegui/blob/main/nicegui/static/utils/dynamic_properties.js and https://github.com/zauberzeug/nicegui-highcharts/blob/main/nicegui_highcharts/highchart.js#L53C9-L53C33

**Reply by evnchn** [reply]
Mentioned in https://github.com/zauberzeug/nicegui/wiki/Advanced-operations-and-techniques#uinotification-with-actions

Let me edit with the reason it works.
